### ISSUE TITLE
FR25 Enable data export or download

### ISSUE THREAD

**Issue by gubuntu** [kartoza/LEDET_BIMS#32]
Allow a user with the appropriate permissions to download the raw data from a map layer

constrained by:
- the view extent
- the user-defined filter

with these format options:
- csv
- ods
- geopackage

nice to have constraints:
- a user-defined polygon
- a selected polygon from another layer

**Comment by gubuntu** [comment]
GeoNode (through Geoserver) provides most of these download formats out of the box for all map layers, so the client should just need to contruct the url with the appropriate filter.

**Comment by meomancer** [comment]
hi @gubuntu 
i'm quite confused for this ticket how to communicate with geonode.
The records data is already in bims, and should the download data process is just creating csv or other format through django bims without geonode?

**Comment by gubuntu** [comment]
yes the biodiversity records download can come from BIMS django.

But downloads of any other layer should come from geonode/geoserver. 

(It should even be possible to publish the biodiversity records in geonode so you can download them from there (and so authorised users can fetch them as WMS or WFS layers) but that should probably go into the baacklog as a separate out of scope issue...)

**Comment by meomancer** [comment]
Currently, it is just csv, due to i'm still having blocker to put geonode data on my localhost

**Comment by gubuntu** [comment]
@meomancer please merge this into GeoNode, then you can use it here. It's been waiting to be merged for ages as part of the GeoNode/geoSAFE project: https://github.com/kartoza/geonode/issues/150

**Comment by Vries14** [comment]
csv allows for downloading raw data in view extent, boundary and filters
